src/main/java/org/maveric/quarkus/panache/services/SavingAccountServices.java

```diff
@@ -114,6 +115,51 @@ else if (isNumeric(search)) {
     }
 
 
+    public ResponseDto getSavingAccountDetailBasedOnAccountId(Long accountId)
+    {
+
+        log.info("Request  ::  accountId {}", accountId);
+        ResponseDto responseDto = null;
+        try {
+            SavingAccount savingAccount = savingAccountRepository.findBySavingsAccountId(accountId);
+
+            if (savingAccount == null) {
+                log.error("Saving account detail not present in db");
+                throw new SavingDetailsNotFoundException("Saving account details not found for this id " + accountId);
+            }
+
+            responseDto = getResponseStructure(SUCCESS_MSG, HttpResponseStatus.OK.code(),
+                    UPDATED_SUCCESS_RESPONSE_MSG, savingAccount, SAVING_ACCOUNTS_URL_PATH);
+            log.info("Response :: {} ", responseDto);
+        } catch (Exception e) {
+            log.error("error :: " + e.getMessage());
+            throw e;
```

Comment:
throw a custom exception like SavingsAccountException instead of throwing it as it is.